cylc/flow/task_events_mgr.py

```diff
@@ -587,8 +587,16 @@ def process_message(
         # Any message represents activity.
         self.reset_inactivity_timer_func()
 
-        if not self._process_message_check(
-                itask, severity, message, event_time, flag, submit_num):
+        with self.workflow_db_mgr.get_pri_dao() as pri_dao:
+            message_already_received = pri_dao.message_in_db(
+                submit_num, message, event_time, itask)
```

Comment:
This context manager automatically closes the DB. But do we need to do that here? cc @oliver-sanders 

If not it can just be accessed like
```suggestion
        message_already_received = self.workflow_db_mgr.pri_dao.message_in_db(
            submit_num, message, event_time, itask)
```

Reply:
Yes, we should not be closing the DB here.

No need to use the context manager when accessing the DB from the Scheduler process itself, the Scheduler has built-in logic to manage DB connections.